src/Serialization/main.jl

```diff
@@ -1,6 +1,20 @@
 using JSON
 using UUIDs
 
+# type that doesn't serialize with backref
+# and stores type outside of vector entries
+# when serialized in a Vector{OscarBasictype}
+OscarBasicType = Union{
+    fmpz,
+    fmpq,
+    Bool,
+    Number,
+    Symbol,
+}
+
+is_basic_serialization_type(::Type) = false
+is_basic_serialization_type(::Type{T}) where T <: OscarBasicType = isconcretetype(T)
```

Comment:
This could be replaced by
```suggestion
is_basic_serialization_type(::Type{fmpz}) = true
is_basic_serialization_type(::Type{fmpq}) = true
is_basic_serialization_type(::Type{Bool}) = true
is_basic_serialization_type(::Type{Symbol}) = true
is_basic_serialization_type(::Type{T}) where T <: Number = isconcretetype(T)
```
And then these methods could (if you feel it makes sense) be moved right next to the code which defines the "basic" serialization methods in each case.